dynamic/provider_test.go

```diff
@@ -73,115 +72,189 @@ func TestPrimitiveTypes(t *testing.T) {
+			"attrIntRequired":             resource.NewProperty(64.0),
+			"attrStringDefaultOverridden": resource.NewProperty("overridden"),
+		}),
+		OldInputs: inputs(),
+	}))
+
+	t.Run("diff(all)", assertGRPCCall(grpc.Diff, &pulumirpc.DiffRequest{
+		Id:  "example-id-2",
+		Urn: urn,
+		Olds: marshal(resource.PropertyMap{
+			"attrBoolComputed":   resource.NewProperty(false),
+			"attrBoolRequired":   resource.NewProperty(true),
+			"attrIntComputed":    resource.NewProperty(128.0),
+			"attrIntRequired":    resource.NewProperty(64.0),
+			"attrNumberRequired": resource.NewProperty(12.3456),
+			"attrStringComputed": resource.NewProperty("t"),
+			"attrStringRequired": resource.NewProperty("s"),
+			"id":                 resource.NewProperty("example-id"),
+		}),
+		News: marshal(resource.PropertyMap{
+			"attrBoolRequired":   resource.NewProperty(false),
+			"attrStringRequired": resource.NewProperty("u"),
+			"attrIntRequired":    resource.NewProperty(65.0),
+			"attrNumberRequired": resource.NewProperty(12.3456789),
+		}),
+		OldInputs: inputs(),
+	}))
+
+	t.Run("delete", assertGRPCCall(grpc.Delete, &pulumirpc.DeleteRequest{
+		Id:         "example-id-delete",
+		Urn:        urn,
+		Properties: outputs(),
+	}))
+
+	t.Run("update", assertGRPCCall(grpc.Update, &pulumirpc.UpdateRequest{
+		Id:   "example-update-id",
+		Urn:  urn,
+		Olds: outputs(),
+		News: marshal(with(outputProps(), resource.PropertyMap{
+			"attrBoolRequired": resource.NewProperty(false),
+		})),
+	}))
+
+	t.Run("read", assertGRPCCall(grpc.Read, &pulumirpc.ReadRequest{
+		Id:         "example-read-id",
+		Urn:        urn,
+		Properties: outputs(),
+	}))
+
+	t.Run("import", assertGRPCCall(grpc.Read, &pulumirpc.ReadRequest{
+		Id:  "example-read-id",
+		Urn: urn,
+	}))
+}
 
-	t.Run("create(preview)", func(t *testing.T) {
-		resp, err := grpc.Create(ctx, &pulumirpc.CreateRequest{
-			Preview:    true,
-			Urn:        urn,
-			Properties: inputs(),
-		})
-		require.NoError(t, err)
-		assertGRPC(t, resp)
-	})
+func TestConfigure(t *testing.T) {
+	t.Parallel()
+	skipWindows(t)
 
-	t.Run("create", func(t *testing.T) {
-		resp, err := grpc.Create(ctx, &pulumirpc.CreateRequest{
-			Urn:        urn,
-			Properties: inputs(),
-		})
-		require.NoError(t, err)
-		assertGRPC(t, resp)
-	})
+	s := grpcTestServer(context.Background(), t)
 
-	t.Run("diff(none)", func(t *testing.T) {
-		resp, err := grpc.Diff(ctx, &pulumirpc.DiffRequest{
-			Id:        "example-id-0",
-			Urn:       urn,
-			Olds:      outputs(),
-			News:      inputs(),
-			OldInputs: inputs(),
-		})
-		require.NoError(t, err)
-		assertGRPC(t, resp)
-	})
+	t.Run("parameterize", assertGRPCCall(s.Parameterize, &pulumirpc.ParameterizeRequest{
+		Parameters: &pulumirpc.ParameterizeRequest_Args{
+			Args: &pulumirpc.ParameterizeRequest_ParametersArgs{
+				Args: []string{pfProviderPath(t)},
+			},
+		},
+	}, noParallel))
+
+	t.Run("check-config", assertGRPCCall(s.CheckConfig, &pulumirpc.CheckRequest{
```

Comment:
These would be a lot more readable if the result being compared are in the same place - the recorded files are all a few lines long, so can be easily inlined.

Reply:
We can do that. I inlined this set of test assertions.